From c406ee1f4697af53ee7e63a592759731e0d35dab Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Thu, 26 Mar 2015 09:28:46 +0000
Subject: [PATCH 3/4] Fix image orientation

Image re-orientation is currently inverted -portrait images are shown upside-down.

diff --git a/mythtv/libs/libmythui/mythimage.cpp b/mythtv/libs/libmythui/mythimage.cpp
index f2785ce..7b951a6 100644
--- a/mythtv/libs/libmythui/mythimage.cpp
+++ b/mythtv/libs/libmythui/mythimage.cpp
@@ -144,25 +144,25 @@ void MythImage::Orientation(int orientation)
         Assign(mirrored(false, true));
         break;
 
-    case 5: // The image is transposed (rotated 90° CW flipped horizontally)
+    case 5: // The image is transposed (flipped horizontally, then rotated 90° CCW)
         matrix.rotate(90);
         Assign(transformed(matrix, Qt::SmoothTransformation));
         Assign(mirrored(true, false));
         break;
 
     case 6: // The image is rotated 90° CCW
-        matrix.rotate(270);
+        matrix.rotate(90);
         Assign(transformed(matrix, Qt::SmoothTransformation));
         break;
 
-    case 7: // The image is transversed  (rotated 90° CW and flipped vertically)
-        matrix.rotate(90);
+    case 7: // The image is transversed (flipped horizontally, then rotated 90° CW)
+        matrix.rotate(270);
         Assign(transformed(matrix, Qt::SmoothTransformation));
-        Assign(mirrored(false, true));
+        Assign(mirrored(true, false));
         break;
 
     case 8: // The image is rotated 90° CW
-        matrix.rotate(90);
+        matrix.rotate(270);
         Assign(transformed(matrix, Qt::SmoothTransformation));
         break;
 
-- 
1.9.1

